setup.py

```diff
@@ -42,6 +42,10 @@
 extensions = None
 cmd_class = {}
 
+DISABLE_COMPILE_ARROW_EXTENSIONS = os.environ.get(
+    "DISABLE_COMPILE_ARROW_EXTENSIONS", False
+)
```

Comment:
Be careful, your code returns a `bool | str`. why not be explicit? Also follow the new env variable prefix
```suggestion
SNOWFLAKE_DISABLE_COMPILE_ARROW_EXTENSIONS = os.environ.get(
    "SNOWFLAKE_DISABLE_COMPILE_ARROW_EXTENSIONS", "false"
).lower() in ("y", "yes", "t", "true", "1", "on")
```